Mark HTTP support as deprecated to be removed in a future release.

johnewart opened the pull request:
# Description

In conversation with @artursouza earlier today regarding issue #763 we came to the conclusion that it might just make sense to simply deprecate the HTTP transport and let people implement their own HTTP client in the cases where they don't want to use gRPC. This would eliminate the split-dependency issue and (I believe) also make the JVM transition easier. 

This PR marks the HTTP-related client constructors, constants and references as deprecated (at least I think I got all the places we instantiate HTTP transport related bits) 

## Issue reference

This is related to (and should be able to close) issue #763 -- also related to proposal #794

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [NA] Extended the documentation (Javadoc generation should be automatic)

mukundansundar commented on sdk/src/main/java/io/dapr/config/Properties.java:
```diff
@@ -45,6 +45,8 @@ public class Properties {
   /**
    * Dapr's default use of gRPC or HTTP for Dapr's method invocation APIs.
+   *
+   * @deprecated HTTP support is being deprecated; use gRPC instead. This will be removed in a future release.
```
@johnewart had a discussion with @artursouza .... Let's not have this deprecated as yet. Only for service invocation... We need to decide on the route of that

artursouza replied:
Yes, we still need this, unless we provide a new API that is called "invokeHTTP" or something. Then this one can be deprecated.

johnewart replied:
Sure, but (for my understanding) is it common for one to use gRPC to talk to Dapr but not want to use it for method invocation?

artursouza replied:
When the service is gRPC, yes. When the service is HTTP, we want clients to use HTTP invocation and we have received feedback that users like to have a convenient method for invoking HTTP services. So, we need to keep an HTTP specific method in the client.

johnewart replied:
Removed the deprecation notice from this portion of the code.

mukundansundar commented on sdk/src/main/java/io/dapr/client/DaprClientBuilder.java:
```diff
@@ -172,6 +175,8 @@ private DaprClient buildDaprClientGrpc() {
+   * @deprecated HTTP support is being deprecated; Use the gRPC builder method (buildDaprClientGrpc) instead. This will be removed in a future release.
    */
   private DaprClient buildDaprClientHttp() {
     return new DaprClientHttp(this.daprHttpBuilder.build(), this.objectSerializer, this.stateSerializer);
```
Log a warning that http client is deprecated and will be removed in future release...

johnewart replied:
I had thought about that but we don't have a standard logging mechanism in the client (I looked and it appears we don't log anything except in the tests). I would hesitate to take additional dependencies unless we are going to add logging elsewhere in the client (which might not be a bad idea but we don't currently). 

Thoughts?

artursouza replied:
That is a great point. We don't have any logging in the SDK on purpose right now to avoid bringing dependencies that might conflict with apps' dependencies. I see 2 options here:

1. Depend on SLF4J, so users can still pick their own logging solution as SLF4J is an abstraction. Of course, the version of SLF4J can conflict and users will need to pick one (probably the latest).
2. Use System.out.Println() - I am OK since most users run this on containers or local mode where logs go to STDOUT or STDERR.

johnewart replied:
I'm fine with just printing to STDOUT for the moment -- while most of the universe uses SLF4J, and it's pretty stable, there are enough issues with dependencies in general that it would be great if we could avoid piling on any more.

johnewart replied:
Added a `System.out.println` statement when constructing `DaprClientHttp` objects.

johnewart replied:
Added a `System.out.println` statement when constructing a `DaprClientHttp` object

mukundansundar reviewed (changes requested):
Small changes. 
Can you also create an issue to track the http client deprecation...

This requires a docs change as well

mukundansundar commented:
Please also see the checkstyle errors

artursouza reviewed (changes requested):
In addition, we need to log warning if the HTTP protocol is chosen via the property we provide in the SDK, this way the user will see the warning in their logs. Users will not see a warning in code due to these deprecation annotations because this code is not used directly by the user, we have a builder that handles which instance of the client to build.

johnewart commented:
> In addition, we need to log warning if the HTTP protocol is chosen via the property we provide in the SDK, this way the user will see the warning in their logs. Users will not see a warning in code due to these deprecation annotations because this code is not used directly by the user, we have a builder that handles which instance of the client to build.

I agree but see above, we don't have a logging façade dependency / logging in place in the client; do we want to add a new dependency like slf4j for that?

johnewart commented:
Fixed checkstyle errors, removed one of the depreciation tags as requested and also added signoff -- amended and force pushed to keep things simpler.

johnewart commented:
Created proposal #794 to track this

artursouza commented:
I responded. I see 2 options:

SLF4J or System.out.Println() because this is just for giving warnings for deprecation and not for other use cases (yet).

mukundansundar reviewed (approved):
lgtm

tanvigour commented on sdk/src/main/java/io/dapr/client/DaprClientHttp.java:
```diff
@@ -114,6 +117,7 @@ public class DaprClientHttp extends AbstractDaprClient {
     this.client = client;
     this.isObjectSerializerDefault = objectSerializer.getClass() == DefaultObjectSerializer.class;
     this.isStateSerializerDefault = stateSerializer.getClass() == DefaultObjectSerializer.class;
+    System.out.println("NOTE: DaprClientHttp is deprecated and will be removed in a future release.");
```
NIT, but it might be more noticeable if we change "NOTE" to "deprecation ALERT" or something to the effect that holds attention as something that shouldn't be ignored as a warning

tanvigour reviewed (approved):
LGTM otherwise

mukundansundar commented:
@artursouza Can this be merged?

artursouza commented:
Closing this in favor of https://github.com/dapr/java-sdk/pull/824
